src/rnnlm/rnnlm-lattice-rescoring.cc

```diff
@@ -0,0 +1,122 @@
+// rnnlm/rnnlm-lattice-rescoring.cc
+
+// Copyright 2017 Johns Hopkins University (author: Daniel Povey) 
+//           2017 Yiming Wang
+//           2017 Hainan Xu
+//
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include <utility>
+
+#include "rnnlm/rnnlm-lattice-rescoring.h"
+#include "util/stl-utils.h"
+#include "util/text-utils.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+KaldiRnnlmDeterministicFst::~KaldiRnnlmDeterministicFst() {
+  int size = state_to_rnnlm_state_.size();
+  for (int i = 0; i < size; i++) {
```

Comment:
no need for the braces.   We'd normally either use int32 or size_t here though- just because int32 is the default type for integers and size_t is the actual type of size().